src/javaRosa/plugin.js

```diff
@@ -673,7 +678,7 @@ define([
                     return mug.options.isSpecialGroup ? 'notallowed' : 'optional';
                 },
                 lstring: gettext('Validation Message'),
-                widget: function (mug, options) {
+                widget: trackLogicRefs(function (mug, options) {
```

Comment:
Unfortunately more indirection here, and it kinda fragile. We cannot set `widget.trackLogicReferences` on the base widget since widgets use a functional style where each derived widget wraps the function/constructor of the widget it's wrapping, effectively hiding all attributes on the inner function/constructor.

@orangejenny this is a disadvantage to the new functional class convention we just adopted (avoiding the `new` operator). However, I think it's worth working around it in a few cases like this rather than deal with the other shortcomings of `new`.

Reply:
Yeah, I see your point. FWIW, I'm not so opposed to `new` in vellum or to generally having slightly different conventions in vellum versus HQ. I'm not categorically opposed to `new` - I don't like it but think it can be okay if used consistently. It worries me in HQ because we use a mix of inheritance styles and are sometimes sloppy about doing `var self = this` in a context where `this` isn't what you'd expect, so it's easy for someone to copy/paste bad code. In vellum the code is so much more consistent that I don't think that's as high of a risk.